optuna/storages/_rdb/storage.py

```diff
@@ -590,135 +578,136 @@ def _update_trial(
 
         """
 
-        session = self.scoped_session()
-
-        trial_model = (
-            session.query(models.TrialModel)
-            .filter(models.TrialModel.trial_id == trial_id)
-            .one_or_none()
-        )
-        if trial_model is None:
-            session.rollback()
-            raise KeyError(models.NOT_FOUND_MSG)
-        if trial_model.state.is_finished():
-            session.rollback()
-            raise RuntimeError("Cannot change attributes of finished trial.")
-        if (
-            state
-            and trial_model.state != state
-            and state == TrialState.RUNNING
-            and trial_model.state != TrialState.WAITING
-        ):
-            session.rollback()
-            return False
+        with _scoped_session(self.scoped_session) as session:
+            trial_model = (
+                session.query(models.TrialModel)
+                .filter(models.TrialModel.trial_id == trial_id)
+                .one_or_none()
+            )
+            if trial_model is None:
+                session.rollback()
+                raise KeyError(models.NOT_FOUND_MSG)
+            if trial_model.state.is_finished():
+                session.rollback()
+                raise RuntimeError("Cannot change attributes of finished trial.")
+            if (
+                state
+                and trial_model.state != state
+                and state == TrialState.RUNNING
+                and trial_model.state != TrialState.WAITING
+            ):
+                session.rollback()
```

Comment:
Not sure but I think adding a handler for `finally` in the context manager should take care of this?
```suggestion
```

Reply:
Since there are no insert or update before this line, I think I can remove this line.